src/java/org/apache/cassandra/index/sai/plan/TopKProcessor.java

```diff
@@ -158,8 +163,11 @@ private <U extends Unfiltered, R extends BaseRowIterator<U>, P extends BaseParti
     {
         // priority queue ordered by score in descending order
         Comparator<Triple<PartitionInfo, Row, ?>> comparator;
-        if (queryVector != null)
+        // TODO does this work for complex expressions?
```

Comment:
What kind of complex expressions are you thinking of? If you mean multiple ORDER BY clauses, then no, it doesn't. With #1422 merged, I am planning on cleaning up this method to help make this section a bit simpler.

Reply:
I'll remove the TODO